### ISSUE TITLE
component including '<Provider>' (preact-redux) replaced while rendering with merge

### ISSUE THREAD

**Issue by csbun** [developit/preact#296]
This issue might related to #276 and #260 or might be an issue to [preact-redux](https://github.com/developit/preact-redux).

dependence:

- redux@3.6.0
- preact-render-to-string@3.0.7
- preact@6.0.1
- preact-redux@1.2.0

environment:

- Chrome Version 52

Here is the example:

```html
<body>
   <!-- render from server -->
   <div id="container">${ssrHtml}</div>
</body>
```

```jsx
class ChildComponent extends Component {
  render() {
    return <div>child</div>;
  }
}
const Child = connect(reducer)(ChildComponent);

class App extends Component {
  render() {
    return (<div id="my-app">
      <Provider store={store}>
        <Child />
      </Provider>
    </div>);
  }
}
// delay 7s to see what happen while client-side rendering with merge
setTimeout(function() {
  render(
    <App />,
    document.getElementById('container'),
    document.getElementById('my-app')
  );
}, 7000);
```

While client-side rendering with merge, `<App>` compared with DOM `#my-app`:

```html
<!-- <App> -->                  | <!-- #my-app -->
<div id="my-app">               | <div id="my-app">
  <Provider store={store}>      |   <div>child</div>   <!-- NOT match -->
    <div>child</div>            |
  </Provider>                   |
</div>                          | </div>
```

Here is what we see in Chrome DevTool 7s after server rendering:
![image](https://cloud.githubusercontent.com/assets/1468366/18339135/4bb55576-75d0-11e6-90bd-7723e792d5eb.png)

Full example on [CodePen](https://codepen.io/csbun/pen/yaNYoL). Much clear on the [github-page](https://csbun.github.io/preact-redux-ssr-example/) (`not found image` request twice).

**Comment by developit** [comment]
@csbun Yikes!  This is due to how Preact tracks ownership of high order children.  It looks like there might be a case here for switching to the innermost child owning the generated DOM.

**Comment by developit** [comment]
I've got a fix! 💃

**Comment by csbun** [comment]
Should I update preact or preact-redux?

**Comment by developit** [comment]
Not yet, still testing.

**Comment by developit** [comment]
The latest raft of updates (about to be released as `6.1`) fix this!

**Comment by developit** [comment]
Fix released in `6.1`! https://codepen.io/developit/pen/ozZYPK

**Comment by csbun** [comment]
Great!
